spark/v3.5/spark/src/main/java/org/apache/spark/sql/catalyst/analysis/IcebergAnalysisException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.spark.sql.catalyst.analysis;
+
+import org.apache.spark.QueryContext;
+import org.apache.spark.sql.AnalysisException;
+import scala.Option;
+import scala.collection.immutable.Map$;
+
+public class IcebergAnalysisException extends AnalysisException {
+  public IcebergAnalysisException(String message) {
+    super(
+        message,
+        Option.empty(),
```

Comment:
We can remove all these non-mandatory arguments

Reply:
Thank you for your comment, but I don’t think I can remove these parameters. If I do, I encounter a compilation error.

Reply:
Hm i tried this and it seemed to compile, i wonder if we have different settings?

```
public class IcebergAnalysisException extends AnalysisException {
  public IcebergAnalysisException(String message) {
    super(message, Map$.MODULE$.<String, String>empty());
  }
}
```

Reply:
OK synced with @huaxingao looks like its not the right method.